go-controller/pkg/util/pod_annotation_unit_test.go

```diff
@@ -35,6 +36,31 @@ func TestMarshalPodAnnotation(t *testing.T) {
 			},
 			expectedOutput: map[string]string{"k8s.ovn.org/pod-networks": `{"default":{"ip_addresses":["192.168.0.5/24"],"mac_address":"","ip_address":"192.168.0.5/24"}}`},
 		},
+		{
+			desc: "single IP assigned to pod with MAC, Gateway, SkipIpConfig and Routes NOT SPECIFIED",
+			inpPodAnnot: PodAnnotation{
+				IPs:          []*net.IPNet{ovntest.MustParseIPNet("192.168.0.5/24")},
+				SkipIPConfig: pointer.Bool(true),
+			},
+			expectedOutput: map[string]string{"k8s.ovn.org/pod-networks": `{"default":{"ip_addresses":["192.168.0.5/24"],"mac_address":"","ip_address":"192.168.0.5/24","skip_ip_config":true}}`},
+		},
+		{
+			desc:        "single IP assigned to pod with MAC, Gateway, Routes NOT SPECIFIED and skip_ip_config at annotations",
+			annotations: map[string]string{"k8s.ovn.org/pod-networks": `{"default":{"skip_ip_config":true}}`},
+			inpPodAnnot: PodAnnotation{
+				IPs: []*net.IPNet{ovntest.MustParseIPNet("192.168.0.5/24")},
+			},
+			expectedOutput: map[string]string{"k8s.ovn.org/pod-networks": `{"default":{"ip_addresses":["192.168.0.5/24"],"mac_address":"","ip_address":"192.168.0.5/24","skip_ip_config":true}}`},
+		},
+		{
+			desc:        "single IP assigned to pod with MAC, Gateway, Routes NOT SPECIFIED and skip_ip_config at annotations and struct",
```

Comment:
this test description could be more descriptive. Something like struct overrides existing annotation value or something like that.